spring-aot/src/main/java/org/springframework/data/RepositoryDefinitionConfigurationProcessor.java

```diff
@@ -239,6 +259,16 @@ void writeConfiguration(RepositoryConfiguration configuration) {
 			writeQueryMethodConfiguration(configuration);
 			writeRepositoryFragments(configuration);
 			writeCustomImplementation(configuration);
+			writeCustomRepositoryBaseClass(configuration);
+		}
+
+		private void writeCustomRepositoryBaseClass(RepositoryConfiguration configuration) {
+
+			if (!configuration.hasCustomBaseClass()) {
+				return;
+			}
+
+			registry.reflection().forType(configuration.getCustomRepositoryBaseClass()).withAccess(TypeAccess.DECLARED_CONSTRUCTORS, TypeAccess.DECLARED_METHODS);
```

Comment:
What about adding `TypeAccess.PUBLIC_METHODS` in order to have also the inherited ones?

Reply:
just public ones should actually do.